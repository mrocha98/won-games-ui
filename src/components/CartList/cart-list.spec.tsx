import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import { CartList } from '.'
import mockItems from './mock'

jest.mock('components/GameItem', () => ({
  __esModule: true,
  GameItem: function Mock() {
    return <div data-testid="game-item" />
  }
}))

describe('<CartList />', () => {
  it('should render the cart list', () => {
    const { container } = renderWithTheme(
      <CartList items={mockItems} total="R$ 330,00" />
    )

    expect(screen.getAllByTestId('game-item')).toHaveLength(2)
    expect(screen.getByText('R$ 330,00')).toHaveStyle({ color: '#f231a5' })

    expect(container.firstChild).toMatchSnapshot()
  })
})
