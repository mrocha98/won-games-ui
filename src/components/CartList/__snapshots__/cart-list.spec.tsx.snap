// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartList /> should render the cart list 1`] = `
.c0 {
  background: #FAFAFA;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-self: start;
  -ms-flex-item-align: start;
  align-self: start;
}

.c1 {
  background: #eee;
  color: #030517;
  font-weight: 600;
  font-size: 1.4rem;
  padding: 2rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c2 {
  color: #F231A5;
}

@media (min-width:768px) {
  .c1 {
    font-size: 1.6rem;
    padding: 2.4rem;
  }
}

<div
  class="c0"
>
  <div
    data-testid="game-item"
  />
  <div
    data-testid="game-item"
  />
  <footer
    class="c1"
  >
    Total 
    <span
      class="c2"
    >
      R$ 330,00
    </span>
  </footer>
</div>
`
